mungegithub/mungers/mungerutil/time_cache_test.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungerutil
+
+import (
+	"testing"
+	"time"
+)
+
+type testLTG struct {
+	number int
+	label  string
+	time   time.Time
+}
+
+func (t *testLTG) Number() int { return t.number }
+func (t *testLTG) FirstLabelTime(label string) *time.Time {
+	if label == t.label {
+		return &t.time
+	}
+	return nil
+}
+
+func TestFirstLabelCache(t *testing.T) {
+	timeA := time.Now()
+	timeB := timeA.Add(time.Minute)
+	table := []struct {
+		obj    testLTG
+		expect time.Time
+	}{
+		{testLTG{1, "lgtm", timeA}, timeA},
+		{testLTG{2, "blah", timeA}, time.Time{}},
+		{testLTG{1, "lgtm", timeB}, timeA}, // because it will be cached
+		{testLTG{2, "lgtm", timeB}, timeB}, // the empty time was not cached.
+	}
+
+	cache := NewLabelTimeCache("lgtm")
+	for i, tt := range table {
+		got, ok := cache.FirstLabelTime(&tt.obj)
+		if e, a := (tt.expect != time.Time{}), ok; e != a {
```

Comment:
oh, it's just a common pattern to avoid typing long variables multiple
times. not needed here, I can remove if you like

On Thu, Aug 18, 2016 at 4:16 PM, Eric Paris <notifications@github.com>
wrote:

> In mungegithub/mungers/mungerutil/time_cache_test.go
> <https://github.com/kubernetes/contrib/pull/1563#discussion_r75406061>:
>
> > +	timeA := time.Now()
> > +	timeB := timeA.Add(time.Minute)
> > +	table := []struct {
> > +		obj    testLTG
> > +		expect time.Time
> > +	}{
> > +		{testLTG{1, "lgtm", timeA}, timeA},
> > +		{testLTG{2, "blah", timeA}, time.Time{}},
> > +		{testLTG{1, "lgtm", timeB}, timeA}, // because it will be cached
> > +		{testLTG{2, "lgtm", timeB}, timeB}, // the empty time was not cached.
> > +	}
> > +
> > +	cache := NewLabelTimeCache("lgtm")
> > +	for i, tt := range table {
> > +		got, ok := cache.FirstLabelTime(&tt.obj)
> > +		if e, a := (tt.expect != time.Time{}), ok; e != a {
>
> what's the point of a in all this? why not just use ok ? I feel like I'm
> missing something...
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/contrib/pull/1563/files/aee73d1142f4fa0657f7e5338b23f4e4ee5f2b3c#r75406061>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAnglmdnEDSTYXsrcLQyhCu963uHyOnGks5qhOe3gaJpZM4Jn-Mc>
> .
>